python/ray/tests/test_gcs_fault_tolerance.py

```diff
@@ -647,6 +648,93 @@ def pid(self):
         ray.get_actor("A")
 
 
+@pytest.fixture
+def redis_replicas(monkeypatch):
+    monkeypatch.setenv("TEST_EXTERNAL_REDIS_REPLICAS", "3")
+
+
+@pytest.mark.parametrize(
+    "ray_start_cluster_head_with_external_redis",
+    [
+        generate_system_config_map(
+            gcs_failover_worker_reconnect_timeout=20,
+            gcs_rpc_server_reconnect_timeout_s=60,
+            gcs_server_request_timeout_seconds=10,
+            redis_db_connect_retries=50,
+        )
+    ],
+    indirect=True,
+)
+def test_redis_failureover(redis_replicas, ray_start_cluster_head_with_external_redis):
+    cluster = ray_start_cluster_head_with_external_redis
+    import redis
```

Comment:
why delayed import?

Reply:
no specific reason, just feel it's only used by this test.